RileyLinkKit/ViewControllers/RileyLinkDeviceTableViewController.swift

```diff
@@ -0,0 +1,353 @@
+        }
+    }
+
+    public override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        switch Section(rawValue: indexPath.section)! {
+        case .Commands:
+            switch CommandRow(rawValue: indexPath.row)! {
+            case .Tune:
+                let vc = CommandResponseViewController(command: { [unowned self] (completionHandler) -> String in
+                    self.device.tunePumpWithResultHandler({ (response) -> Void in
+                        switch response {
+                        case .Success(let scanResult):
+                            var resultDict: [String: AnyObject] = [:]
+                            let decimalFormatter = NSNumberFormatter()
+                            decimalFormatter.minimumSignificantDigits = 5
+
+                            resultDict["Best Frequency"] = scanResult.bestFrequency
+                            resultDict["Trials"] = scanResult.trials.map({ (trial) -> String in
+                                return "\(decimalFormatter.stringFromNumber(trial.frequencyMHz)!) MHz  \(trial.successes)/\(trial.tries)  \(trial.avgRSSI)"
+                            })
+
+                            var responseText: String
+
+                            if let data = try? NSJSONSerialization.dataWithJSONObject(resultDict, options: .PrettyPrinted), string = String(data: data, encoding: NSUTF8StringEncoding) {
+                                responseText = string
+                            } else {
+                                responseText = "No response"
+                            }
+
+                            completionHandler(responseText: responseText)
+                        case .Failure(let error):
+                            completionHandler(responseText: String(error))
+                        }
+                    })
+
+                    return "Tuning radio..."
+                })
+
+                vc.title = "Tune device radio"
+
+                self.showViewController(vc, sender: indexPath)
+            case .ChangeTime:
+                let vc = CommandResponseViewController { [unowned self] (completionHandler) -> String in
+                    self.device.syncPumpTime { (error) -> Void in
+                        dispatch_async(dispatch_get_main_queue()) {
+                            if let error = error {
+                                completionHandler(responseText: "Failed: \(error)")
+                            } else {
+                                completionHandler(responseText: "Succeeded")
+                            }
+                        }
+                    }
+
+                    return "Changing time..."
+                }
+
+                vc.title = "Change Time"
+
+                self.showViewController(vc, sender: indexPath)
+            case .MySentryPair:
+                let vc = self.storyboard!.instantiateViewControllerWithIdentifier("mySentryPair") as! MySentryPairViewController
```

Comment:
Also I think we might be able to get rid of the custom MySentryPairViewController altogether since it turns out the MySentry ID didn't end up being useful at all. `CommandResponseViewController` could start with the pairing instructions as it starts pinging.